libcodechecker/server/api/report_server.py

```diff
@@ -713,72 +698,107 @@ def getRunResults(self, run_ids, limit, offset, sort_types,
-                                           File.filename.label('filename')) \
-                    .outerjoin(File, report_query.c.file_id == File.id)
-
-                file_query = self.__sortResultsQuery(
-                    file_query, sort_types, is_unique, report_query)
-
-                for report_hash, checker, checker_msg, severity, file_id, \
-                    rs_status, rs_msg, rs_author, rs_date, path in \
-                        file_query:
-
-                    review_data = None
-                    if rs_status:
-                        review_data = ReviewData(
-                            status=review_status_enum(rs_status),
-                            comment=rs_msg,
-                            author=rs_author,
-                            date=str(rs_date))
-                    else:
-                        review_data = ReviewData(
-                            status=ttypes.ReviewStatus.UNREVIEWED)
+                               ReviewStatus.bug_hash == Report.bug_id) \
+                    .outerjoin(sorted_reports,
+                               sorted_reports.c.id == Report.id) \
+                    .filter(sorted_reports.c.id.isnot(None))
+
+                for report_id, bug_id, checker_msg, checker, severity, \
+                    status, filename, path in \
+                        q:
+                    review_data = create_review_data(status)
 
                     results.append(
-                        ReportData(bugHash=report_hash,
-                                   checkedFile=path,
+                        ReportData(bugHash=bug_id,
+                                   checkedFile=filename,
                                    checkerMsg=checker_msg,
                                    checkerId=checker,
                                    severity=severity,
                                    reviewData=review_data))
             else:
-                q = self.__sortResultsQuery(q, sort_types)
+                q = session.query(Report.id, Report.file_id, Report.line,
+                                  Report.column, Report.detection_status,
+                                  Report.bug_id, Report.checker_message,
+                                  Report.checker_id, Report.severity,
+                                  ReviewStatus, File.filepath) \
+                    .outerjoin(File, Report.file_id == File.id) \
+                    .outerjoin(ReviewStatus,
+                               ReviewStatus.bug_hash == Report.bug_id) \
+                    .filter(filter_expression)
 
-                for report, source_file, review_status in \
-                        q.limit(limit).offset(offset):
+                if run_ids:
+                    q = q.filter(Report.run_id.in_(run_ids))
 
-                    review_data = create_review_data(review_status)
+                if cmp_data:
+                    q = q.filter(Report.bug_id.in_(diff_hashes))
```

Comment:
Do we use the file name when calculating the diff between two runs?

Reply:
No, we don't use the file name in diff. We use only bug hash.